### PR TITLE
Implement fsm

### PR DESCRIPTION
**Pull request by ChristienGuy**

### CODE REVIEW

File: src/gameStateMachine.ts [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+interface Lyrics {
+    warped: Lyric[];
+}
+
+interface Song {
+    id: number;
+    title: string;
+    artistName: string;
+    lyrics: Lyrics;
+}
```
Comment (Jack-Gill) [inline src/gameStateMachine.ts:22]: These are used in the page I'm working on at the moment, once we have this one merged to master and I've pulled it into my feature branch I'll have a look at putting them somewhere shared!

File: src/gameStateMachine.ts [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+    | {
+          type: "SELECT_ARTIST";
+          artist: Artist;
+      }
+    | {
+          type: "CLEAR_SELECTED_ARTIST";
+      }
+    | {
+          type: "RESOLVE";
+      }
```
Comment (Jack-Gill) [inline src/gameStateMachine.ts:45]: You mentioned removing this one, have you just not pushed your latest?

Comment (ChristienGuy) [reply src/gameStateMachine.ts:45]: ah, I removed it from the machine, but didn't remove the typing for the event.

File: src/gameStateMachine.ts (hunk 1 above)
Comment (ChristienGuy) [reply src/gameStateMachine.ts:22]: Yeah for shared types we can move them into just a big `types.ts` file at the root. Seems to be relatively standard and good enough for us atm.

File: functions/proxy/proxy.js [hunk 3]
```diff
@@ -2,19 +2,20 @@
...
+    const url = new URL(`https://genius.com/api/${urlString}`);
     url.search = new URLSearchParams(event.queryStringParameters);
 
     try {
         const response = await fetch(url, {
             headers: {
                 Accept: "application/json",
-                Authorization: `Bearer ${GENIUS_AUTH_TOKEN}`
-            }
+                // Authorization: `Bearer ${GENIUS_AUTH_TOKEN}`,
```
Comment (Jack-Gill) [inline functions/proxy/proxy.js:17]: Do we not need this token any more?

Comment (ChristienGuy) [reply functions/proxy/proxy.js:17]: Oh yeah, this was me messing with the different API. Which doesn't appear to need a token but also isn't really public, not sure what the best option here is but the new one lets you search by artist and song separately.

Comment (Jack-Gill) [reply functions/proxy/proxy.js:17]: Ah okay, well I guess innocent until proven guilty! Let's mess with the new one until we get stopped by the fuzz.

**Pull request merged**